### PR TITLE
Changes for removal of Dart_ReadKernelBinary and refactor build targets.

### PR DESCRIPTION
**Pull request by rmacnak-google**

### CODE REVIEW

File: runtime/dart_vm.cc [hunk 1]
```diff
@@ -340,11 +328,11 @@ DartVM::DartVM(const Settings& settings,
       Dart_IsDart2Snapshot(isolate_snapshot_->GetData()->GetSnapshotPointer());
 
   const bool is_preview_dart2 =
-      platform_kernel_ != nullptr || isolate_snapshot_is_dart_2;
+      (platform_kernel_mapping_->GetSize() > 0) || isolate_snapshot_is_dart_2;
```
Comment (a-siva) [inline runtime/dart_vm.cc]: GetPlatformKernel().GetSize() > 0

File: runtime/dart_vm.cc [hunk 2]
```diff
@@ -340,11 +328,11 @@ DartVM::DartVM(const Settings& settings,
...
 
   FXL_DLOG(INFO) << "Dart 2 " << (is_preview_dart2 ? "is" : "is NOT")
                  << " enabled. Platform kernel: "
-                 << static_cast<bool>(platform_kernel_)
+                 << static_cast<bool>(platform_kernel_mapping_->GetSize() > 0)
```
Comment (a-siva) [inline runtime/dart_vm.cc]: GetPlatformKernel().GetSize() > 0

**Comment by zanderso** [comment]
lgtm. Sorry for the confusion. I didn't think we were depending on libdart_platform anywhere else.

File: shell/testing/BUILD.gn [hunk 3]
```diff
@@ -22,7 +22,7 @@ executable("testing") {
     "//third_party/dart/runtime:libdart_jit",
     "//third_party/dart/runtime/bin:embedded_dart_io",
     "//third_party/dart/runtime/bin:libdart_builtin",
-    "//third_party/dart/runtime/platform:libdart_platform",
```
Comment (zanderso) [inline shell/testing/BUILD.gn]: I had removed this without any problem in my local testing. Does it need to be there in the Fuchsia build?

**Pull request merged**